BibBuddy/app/src/main/java/de/bibbuddy/HomeFragment.java

```diff
@@ -1,38 +1,151 @@
 package de.bibbuddy;
 
+import android.content.Context;
 import android.os.Bundle;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
+import android.widget.TextView;
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
-import androidx.fragment.app.Fragment;
+import androidx.recyclerview.widget.RecyclerView;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ThreadLocalRandom;
 
 /**
- * The HomeFragment is responsible for TODO.
+ * The HomeFragment is responsible for welcoming the user and displaying recently used books.
  *
- * @author Claudia Schönherr
+ * @author Claudia Schönherr, Luis Moßburger
  */
-public class HomeFragment extends Fragment {
+public class HomeFragment extends BackStackFragment
+    implements BookRecyclerViewAdapter.BookListener {
 
   private View view;
+  private Context context;
+
+  private BookDao bookDao;
+  private BookModel bookModel;
+  private DatabaseHelper dbHelper;
+
+  private int bookAmount;
+  List<BookItem> bookItemList;
+  private BookRecyclerViewAdapter adapter;
 
   @Nullable
   @Override
   public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                            @Nullable Bundle savedInstanceState) {
 
     view = inflater.inflate(R.layout.fragment_home, container, false);
+    context = view.getContext();
 
     MainActivity mainActivity = (MainActivity) requireActivity();
     mainActivity.setVisibilityImportShareButton(View.GONE, View.GONE);
     mainActivity.setVisibilitySortButton(false);
 
     mainActivity.updateHeaderFragment(getString(R.string.navigation_home));
     mainActivity.updateNavigationFragment(R.id.navigation_home);
+    updateWelcomeMessage();
+
+    bookModel = new BookModel(context, 1L);
+    dbHelper = new DatabaseHelper(context);
+    bookDao = new BookDao(dbHelper);
+
+    bookAmount = 3;
+    List<Book> bookList = bookDao.findModifiedBooks(bookAmount);
```

Comment:
```suggestion
    List<Book> bookList = bookModel.findModifiedBooks(bookAmount);
```
I suggest to create a bridge method (that simply call the used bookDao-methods as connection).

Reply:
Thanks, done!

Reply:
Thanks, done!